ethereum_history_api/circuits/lib/src/rlp/decode.nr

```diff
@@ -57,21 +57,29 @@ pub fn decode_to_rlp_header<N>(arr: [u8; N]) -> RlpHeader {
     if (prefix < 0x80) { // single byte
         RlpHeader { offset: 0, length: 1, data_type: STRING }
     } else if (prefix < 0xb8) { // 0-55 byte string
+        let offset = 1;
+        let length = wrapping_sub(prefix, 0x80) as u64;
+        assert(offset + length <= N, "Length of short string exceeds input length");
+
         RlpHeader { offset: 1, length: wrapping_sub(prefix, 0x80) as u64, data_type: STRING }
     } else if (prefix < 0xc0) { // >55-byte string
-        RlpHeader {
-            offset: wrapping_sub(1 + prefix, 0xb7) as u64,
-            length: extract_payload_len(arr, wrapping_sub(prefix, 0xb7) as u64),
-            data_type: STRING
-        }
+        let offset = wrapping_sub(1 + prefix, 0xb7) as u64;
+        let length = extract_payload_len(arr, wrapping_sub(prefix, 0xb7) as u64);
+        assert(offset + length <= N, "Length of long string exceeds input length");
```

Comment:
What do you think about adding "decoded" at the beginning of the assertion message? "Decoded length of long string exceeds input length". I think it might improve readability as without it sounds a bit as if it was real length and not length that we decoded

Reply:
Changed it :)